src/include/mpi.h.in

```diff
@@ -167,7 +167,7 @@ typedef int MPI_Datatype;
 #define MPI_WCHAR          ((MPI_Datatype)@MPI_WCHAR@)
 #define MPI_SHORT          ((MPI_Datatype)@MPI_SHORT@)
 #define MPI_UNSIGNED_SHORT ((MPI_Datatype)@MPI_UNSIGNED_SHORT@)
-#define MPI_INT            ((MPI_Datatype)@MPI_INT@)
+#define MPI_INT            ((MPI_Datatype)0x4c000405)
```

Comment:
Is this a temporary workaround?

Reply:
The intention is to use fixed constants for all builtin MPI datatypes, just as we do in the MPI ABI header. Here we need use 0x4c000405 due to MPICH ABI. I only did MPI_INT to test out. Will apply to the rest next.

Reply:
> Here we need use 0x4c000405 due to MPICH ABI.

IIRC, some of these values were platform dependent. If that is indeed the case, cannot such change potentially break the MPICH ABI in some platforms?

Reply:
Yeah, you are right, but I really don't want to keep all the unused cruft just for the MPICH ABI on different platforms. We'll keep a 64-bit ABI. Hopefully, it'll be practically nonexistent for people to build on 32bit and expecting it to be ABI-swappable with an old 32-bit build today.